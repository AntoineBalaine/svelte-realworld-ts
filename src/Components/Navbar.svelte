<script lang="ts">
	import { page } from "$app/stores";
	import type { PageData } from "../routes/$types";
	export let data: PageData;
	let route: string | null;
	let isLoggedIn: boolean;
	const { user } = data;
	page.subscribe((pageContents) => {
		isLoggedIn = pageContents.data.isLoggedIn;
		route = pageContents.route.id;
	});
</script>

<nav class="navbar navbar-light">
	<div class="container">
		<a class="navbar-brand" href="/">conduit</a>
		<ul class="nav navbar-nav pull-xs-right">
			<li class="nav-item">
				<!-- Add "active" class when you're on that page" -->
				<a class="nav-link active" class:active={!route || route === "/"} href="/">Home</a>
			</li>
			{#if user}
				<li class="nav-item">
					<a class="nav-link" href="/"> <i class="ion-compose" />&nbsp;New Article </a>
				</li>
				<li class="nav-item">
					<a class="nav-link" href="/settings"> <i class="ion-gear-a" />&nbsp;Settings </a>
				</li>
				<li class="nav-item">
					<a class="nav-link" href="/sign-out">Sign out</a>
				</li>
			{:else}
				<li class="nav-item">
					<a class="nav-link" class:active={route === "/sign-in"} href="/sign-in">Sign in</a>
				</li>
				<li class="nav-item">
					<a class="nav-link" class:active={route === "/sign-up"} href="/sign-up">Sign up</a>
				</li>
			{/if}
		</ul>
	</div>
</nav>
