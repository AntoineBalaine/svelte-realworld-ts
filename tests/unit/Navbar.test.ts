import { render } from "@testing-library/svelte";
import { describe, test, expect } from "vitest";
import Navbar from "src/components/Navbar.svelte";

import type { ComponentProps } from "svelte";
import { articleSample } from "tests/DataSamples";

describe("VerticalTabs Component", () => {
	test("should render the component", () => {
		const navbarProps: ComponentProps<Navbar> = {
			data: {
				articles: [articleSample],
				user: null,
				tags: { tags: Array<string>() },
				pages: 1,
				isLoggedIn: false,
				pageNumber: 1
			}
		};
		const navbarPreview = render(Navbar, navbarProps);

		expect(navbarPreview.getByTestId("HomeNav").innerHTML).toEqual("Home");

		expect(navbarPreview.container).toMatchSnapshot();
	});
});
