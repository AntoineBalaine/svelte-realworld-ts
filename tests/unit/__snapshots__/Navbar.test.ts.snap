// Vitest Snapshot v1

exports[`VerticalTabs Component > should render the component 1`] = `
<body>
  <div>
    <nav
      class="navbar navbar-light"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          data-testid="TitleNav"
          href="/"
        >
          conduit
        </a>
         
        <ul
          class="nav navbar-nav pull-xs-right"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link active"
              data-testid="HomeNav"
              href="/"
            >
              Home
            </a>
          </li>
           
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              data-testid="SignInNav"
              href="/sign-in"
            >
              Sign in
            </a>
          </li>
           
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              data-testid="SignUpNav"
              href="/sign-up"
            >
              Sign up
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <!--&lt;Navbar&gt;-->
  </div>
</body>
`;
